packages/@aws-cdk/aws-applicationsignals-alpha/lib/slo/slo-types.ts

```diff
@@ -0,0 +1,138 @@
+import { ISlo, PeriodBasedSloProps, RequestBasedSloProps } from './slo';
+import { ComparisonOperator, MetricType } from './constants';
+import { Goal } from './interval';
+import { PeriodBasedAppSignalsMetricProps, PeriodBasedCloudWatchMetricProps, RequestBasedAppSignalsMetricProps, RequestBasedCloudWatchMetricProps } from './metric';
+import { Resource } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as applicationsignals from 'aws-cdk-lib/aws-applicationsginals';
+
+/**
+ * Base class for all SLO implementations
+ */
+export abstract class ServiceLevelObjective extends Resource implements ISlo {
```

Comment:
We usually follow the `<<CFN Resource Name>>Base` naming pattern. Name this as ServiceLevelObjectiveBase.